core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -47,7 +50,8 @@ abstract class AbstractFetcherThread(name: String,
                                      val sourceBroker: BrokerEndPoint,
                                      fetchBackOffMs: Int = 0,
                                      isInterruptible: Boolean = true,
-                                     includeLogTruncation: Boolean)
+                                     includeLogTruncation: Boolean,
+                                     useLeaderEpochInResponse: Boolean = true)
```

Comment:
My personal opinion is that it may be more general to just pass the `interBrokerProtocolVersion` to the constructor of `AbstactFetcherThread`. And we use this variable to determine the version of OffsetsForLeaderEpochRequest when we actually generate the builder for OffsetsForLeaderEpochRequest. It is more consistent with the existing usage `KafkaConfig.interBrokerProtocolVersion` in the code base. And if in the future there is some other logic that relies on the interBrokerProtocol in the AbstractFetcherThread, we won't need to add more variable to the constructor.

Reply:
I agree. However, I just tried it, and it requires changes to ConsumerFetcherThread constructor, and then ConsumerFetcherManager, and so on. I think it would be easy to change later when we need more logic dependent on inter broker protocol version, and especially once we remove old consumer fetcher code.